script.rb

```diff
@@ -20,32 +20,34 @@ def get_password(prompt = 'Password: ')
 
 def authenticate(page)
   page.form_with(name: nil) do |form|
-    form.username = get_username
-    form.password = get_password
-    puts ''
+    form.username = ENV['ELIXIRSIPS_DPDCART_USERNAME'] || get_username
+    form.password = ENV['ELIXIRSIPS_DPDCART_PASSWORD'] || get_password
   end.submit
 end
 
-def download_video(agent, link)
+def download_file(agent, link)
   post = agent.get BASE_URI + link.uri.to_s
-  video_link = post.links.find { |l| l.text.end_with? '.mp4' }
-  video_name = video_link.text.gsub('_', ' ')
-
-  if File.exists? "./videos/#{video_name}"
-    puts "Already Downloaded #{video_name}"
-    return
+  post.search('//*[@id="blog-container"]/div/ul/li/a').each do |link|
+  file_link, file_name = link['href'], link.text.strip
+  if File.extname(file_name) == '.mkv'
+    puts "Skipping #{file_name}..."
+    next
+  elsif File.exists? "./videos/#{file_name}"
+      puts "Already Downloaded #{file_name}"
+      return
+    end
```

Comment:
Bad indent.

On Aug 31, 2016 11:08 PM, "Ben A. Morgan" <notifications@github.com> wrote:

> In script.rb
> <https://github.com/BenMorganIO/elixirsips/pull/1#discussion_r77082876>:
>
> >    post = agent.get BASE_URI + link.uri.to_s
> > -  video_link = post.links.find { |l| l.text.end_with? '.mp4' }
> > -  video_name = video_link.text.gsub('_', ' ')
> > -
> > -  if File.exists? "./videos/#{video_name}"
> > -    puts "Already Downloaded #{video_name}"
> > -    return
> > +  post.search('//*[@id="blog-container"]/div/ul/li/a').each do |link|
> > +  file_link, file_name = link['href'], link.text.strip
> > +  if File.extname(file_name) == '.mkv'
> > +    puts "Skipping #{file_name}..."
> > +    next
> > +  elsif File.exists? "./videos/#{file_name}"
> > +      puts "Already Downloaded #{file_name}"
> > +      return
> > +    end
>
> extra end?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BenMorganIO/elixirsips/pull/1/files/b1711f5711e54c1fea216e3844323cce1042b3ec#r77082876>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAyOlaXxOTOtgjvDM2j9JM6QcUzhn_4Dks5qlft3gaJpZM4JvaDf>
> .
>